### ISSUE TITLE
JsonAPI verwenden

### ISSUE THREAD

**Issue by mathebox** [Strassengezwitscher/Strassengezwitscher#322]
JsonAPI ist ein [Standard](http://jsonapi.org/) zu Übertragung von "Elementen".
Zum einen wäre dann unsere API sauberer. Zum anderen sollte uns das beim Lösen des Problems helfen, dass @Bensk1 in #205 beschrieben hat.

Für JsonAPI gibt es auch schon die passenden Pakete:
 - Angular2: https://github.com/ghidoz/angular2-jsonapi
 - DRF: https://github.com/django-json-api/django-rest-framework-json-api

Es könnte also mit relativ wenig Aufwand umgesetzt werden.
Meinung? @piepmatz @Bensk1

**Comment by Bensk1** [comment]
Wenn wir damit einfach das Problem lösen können, würde uns das beim Refactoring der MapComponent sehr helfen.

**Comment by piepmatz** [comment]
Ich bin da auch mal drüber gestolpert und finde das grundsätzlich gut.
Allerdings: 

> rest_framework_json_api assumes you are using class-based views in Django Rest Framework.

Wenn ich mich recht erinnere, machen wir das nicht überall. Die Antworten aufs Verschicken des Kontaktformulars z.B. sind glaube ich einfach nur manuell zusammengebaute dicts.
Wenn wir das verwenden wollen, sollten wir es am ehesten überall benutzen.
Da müsste man also womöglich an manchen Stellen noch was umbauen. Vielleicht gibts ja aber auch einen simplen Mechanismus, Fehlermeldungen ans Frontend zu schicken.

Von daher: +1

**Comment by mathebox** [comment]
Okay, ich wusste nicht, dass wir dafür `class-based views` brauchen. Aber es sollte ja kein Problem darstellen das umzuschreiben.

JsonAPI hat auch ein [`errors`-Feld](http://jsonapi.org/format/#errors), um Fehler zurückzugeben. Meinst du sowas, @piepmatz?

DRF-json-api baut auch nicht nur magisch die Response zusammen. Das kann man auch manuell machen. (Hab ich schon mal gemacht.)

Gut dann würde ich mir mal der Issue schnappen.

**Comment by piepmatz** [comment]
Ja, irgendwie sowas wird schon passen für Fehlermeldungen. Müssen halt nur mal schauen, ob auch dafür class-based views benötigt werden oder nicht. Aber daran wirds nicht scheitern.
